src/ploomber/spec/taskspec.py

```diff
@@ -401,12 +403,17 @@ def _init_task(data, meta, project_root, lazy_import, dag):
         task_dict['params'] = resolve_resources(task_dict['params'],
                                                 relative_to=project_root)
 
+    if task_defaults != "":
```

Comment:
the condition here should be to check if we have defaults for the given class:

```
if class_.__name__ in task_defaults:
   ...
```

you can even remove the if statement:


```
init_params = {**tak_defaults.get(class_.__name__ , {}), **task_dict}
```